### DISCUSSION TITLE
When linting, errors like "Component name '[id]' [or `index`] should always be multi-word" are thrown

### DISCUSSION THREAD

**Discussion by rowild** [nuxt/nuxt#20541] [Questions]
Hi!

New to Nuxt 3, but not to view. Trying to implement eslint and prettier, but they give me headaches.
How do you guys handle the above mentioned error, if I may ask? 

`[id]` and `index` pages are pretty much standard if I am not totally wrong. So how would I properly rename an `index` page to concur to the best coding practies? `IndexPage.vue` does not work. Neither does `[id]Page.vue`.
Nuxt itself does not document this scenario: https://nuxt.com/docs/guide/directory-structure/pages/#usage

And so far my google search has not been successful...

Thank you!

**Comment by maou-shonen** [comment]
Although I don't quite understand your question...
But I think what you need is this https://github.com/nuxt/eslint-config

**Reply by AloisSeckar** [reply]
I have this @nuxtjs/eslint-config-typescript dependency in my projects, but having similar "issue". The violated eslint rule is `vue/multi-word-component-names`.

For some reason it is ignoring that Nuxt is composing real component name from its directory structure - so `/components/foo/Bar.vue` becomes `<FooBar>` in application. But eslint is mad, because it only checks the file name and "component name should be multi-word"...

**Reply by maou-shonen** [reply]
It sounds like your issue has been resolved in [this PR](https://github.com/nuxt/eslint-config/pull/262). (unpublished)
If not, maybe you can open a new issue in nuxt/eslint-config?

**Reply by AloisSeckar** [reply]
Yes, that seems like my problem. I will wait till the fix comes out then. Thanks for pointing out for me.

**Comment by AloisSeckar** [comment]
I decided to bypass this by putting `"vue/multi-word-component-names": "off"` inside `rules` section of my `.eslintrc.json` config to hide this rule "violation" until I find some viable solution.

**Comment by rowild** [comment]
Yes, setting this rule to "off" is what I did, too. (And it seems this has to be done with some other rules, too.)
Thanks for the feedback, guys! I close this issue, since another one covers the topic already.
